src/coins.cpp

```diff
@@ -488,19 +505,59 @@ double CCoinsViewCache::GetPriority(const CTransaction &tx, int nHeight, CAmount
 CCoinsViewCursor::~CCoinsViewCursor() {}
 static const size_t nMaxOutputsPerBlock =
     DEFAULT_LARGEST_TRANSACTION / ::GetSerializeSize(CTxOut(), SER_NETWORK, PROTOCOL_VERSION);
-const Coin &AccessByTxid(const CCoinsViewCache &view, const uint256 &txid)
-{
-    // We may return a reference to a coin cache entry so therefore we must make sure the caller
-    // has taken a lock on cs_utxo.
-    AssertLockHeld(view.cs_utxo);
 
+CoinAccessor::CoinAccessor(const CCoinsViewCache &view, const uint256 &txid) : cache(&view), lock(cache->csCacheInsert)
+{
+    cache->cs_utxo.lock_shared();
+    EnterCritical("CCoinsViewCache.cs_utxo", __FILE__, __LINE__, (void *)(&cache->cs_utxo));
     COutPoint iter(txid, 0);
```

Comment:
is the lock stack getting updated properly when doing EnterCritical() and when in DEBUG_LOCKORDER, in the same way it does when we used ENTER_CRITICAL_SECTION()?